scripts/west_commands/completion/west-completion.fish

```diff
@@ -196,21 +196,52 @@ function __zephyr_west_complete_help
 end
 
 function __zephyr_west_complete_board
-    # HWMv1
-    set -l boards (west 2>/dev/null boards --format="{name} {arch}")
-    for board in $boards
-        set -l b (string split " " $board)
-        printf "%s\n" $b[1]\t"$b[2]"
+    set -l one_week_in_s 604800
```

Comment:
One week is a very long time, afaict the cache is not updated even in cases where a new board or variant is added.

Also users switching between upstream and downstream projects, which might have different boards defined, will be hit by the fact that either boards are present, but not completed, or completed but not present in the workspace.

Also this PR doesn't take into consideration the possibility of having multiple workspaces with different releases, for example `zephyr/main` and latest Zephyr LTS.

Reply:
> west can be called outside a git workspace so it's not possible to rely on it. 

Yes, but `west build` is an extension command and needs to be called from within a `west workspace`, although it can be in called in a folder inside the west workspace that is not within a git repository.

Note, #69904 already updated fish completion from calling list_board.py twice, to just call it once, making it comparable to bash and zsh wrt. to execution time. Bash and zsh already called list_boards only once.

> The solution could be to use west instead but then we need to call west every time making the caching useless.

That would actually be worse, cause `west boards` roughly takes 50% longer than `list_boards.py` due to the overhead of extension command handling.

> Also, if you work on adding a new board there is a high chance that you know it's name

I didn't say anything about being the author of a new board.
It could be a new board or an added variant that you get by a `git fetch` or `west update`.

In such a case the user might know the board name, but not the SoC, CPU clusters, or variants, in which case the completion of those when testing the new board is very helpful. But with a long lived cache, then completion becomes useless.

Reply:
> Yes, but west build is an extension command and needs to be called from within a west workspace, although it can be in called in a folder inside the west workspace that is not within a git repository.

You're absolutely right, I forgot that. I updated the PR to update the cache if the latest git commit of Zephyr is updated.

> Note, https://github.com/zephyrproject-rtos/zephyr/pull/69904 already updated fish completion from calling list_board.py twice, to just call it once, making it comparable to bash and zsh wrt. to execution time. Bash and zsh already called list_boards only once.

I saw that, but to be honest it's still slow, bash and zsh could benefit from this approach.

> That would actually be worse, cause west boards roughly takes 50% longer than list_boards.py due to the overhead of extension command handling.

Yes, calling `west boards` is not an option. To get the latest commit I call `west topdir` and then git.

> I didn't say anything about being the author of a new board.
It could be a new board or an added variant that you get by a git fetch or west update.

> In such a case the user might know the board name, but not the SoC, CPU clusters, or variants, in which case the completion of those when testing the new board is very helpful. But with a long lived cache, then completion becomes useless.

That's right, it should be handled by the new approach based on the latest git commit.